x/migration/types/message_claim_morse_supplier.go

```diff
@@ -38,65 +38,72 @@ func NewMsgClaimMorseSupplier(
 }
 
 // ValidateBasic ensures that:
-// - The morseSignature length is valid (signature validation performed elsewhere).
-// - The morseSrcAddress is valid (i.e. it is a valid hex-encoded address).
-// - The shannonDestAddress is valid (i.e. it is a valid bech32 address).
+// - The shannon owner address is valid (i.e. it is a valid bech32 address).
+// - The shannon operator address is valid (i.e. it is a valid bech32 address).
+// - The supplier service configs are valid.
+// - The morsePublicKey is valid.
+// - The morseSrcAddress matches the public key.
+// - The morseSignature is valid.
 func (msg *MsgClaimMorseSupplier) ValidateBasic() error {
-	if len(msg.MorseSignature) != MorseSignatureLengthBytes {
-		return ErrMorseSupplierClaim.Wrapf(
-			"invalid morse signature length; expected %d, got %d",
-			MorseSignatureLengthBytes, len(msg.MorseSignature),
+	if _, err := sdk.AccAddressFromBech32(msg.GetShannonOwnerAddress()); err != nil {
```

Comment:
Should we ensure `OwnerAddress != ""` like below?

Reply:
No, in fact, the check below shouldn't be there either.